src/pages/projectSettings/tabs/GithubCommitQueueTab/getErrors.ts

```diff
@@ -0,0 +1,84 @@
+import { string } from "utils";
+import { AliasFormType, ProjectType } from "../utils";
+
+const { joinWithConjunction } = string;
+
+enum ErrorType {
+  Error,
+  Warning,
+  None,
+}
+
+const getErrorStyle = (
+  errorType: ErrorType,
+  versionControlEnabled: boolean,
+  projectType: ProjectType,
+  fieldName: string
+) => {
+  if (errorType === ErrorType.Warning) {
+    const definitionLocations =
+      projectType === ProjectType.Repo ? ["repo"] : ["project"];
+    if (projectType === ProjectType.AttachedProject) {
+      definitionLocations.push("repo");
+    }
+    if (versionControlEnabled) {
+      definitionLocations.push("Evergreen configuration file");
+    }
+
+    return {
+      "ui:warnings": [
+        `This feature will only run if a ${fieldName} is defined in the ${joinWithConjunction(
+          definitionLocations,
+          "or"
+        )}.`,
+      ],
+    };
+  }
+  if (errorType === ErrorType.Error) {
+    return {
+      "ui:errors": [
+        `A ${fieldName} must be specified for this feature to run.`,
+      ],
+    };
+  }
+  return {};
+};
+
+export const sectionHasError = (
```

Comment:
can we include a return type for this function or getErrorStyle. It's not easy to understand the shape from the ts type inference